resources/components/miningComponent.gd

```diff
@@ -2,33 +2,41 @@ extends Node
 
 class_name MiningComponent
 
-func mine() -> void:
-	pass
+@export var ore_text_scene = preload("res://resources/other/ore_text.tscn")
 
-#	wage idea/direction how mining could be done from the erly tests
+var rng = RandomNumberGenerator.new()
+var min_ore_from_ore := 1
+var max_ore_from_ore := 4
 
-#func _unhandled_input(event: InputEvent) -> void:
-#	if event.is_action_pressed("LMB"):
-#		get_viewport().set_input_as_handled()
-#		var collision :RayCast2D = $RayCast2D
-#		print("Colliding with: " + str(collision.get_collider()))
-#		if collision.is_colliding() and collision.get_collider() is TileMap:
-#			var point := collision.get_collision_point()
-#			print("point" + str(point))
-#			var map :TileMap = collision.get_collider()
-#
-#			var scale := map.scale
-#			print("scale: " + str(scale))
-#			point /= map.scale
-#
-#			var normal = collision.get_collision_normal()
-#			print("normal: " + str(normal))
-#			point -= normal
-#
-#			print("point modified: " + str(point))
-#
-#			var point2 = map.local_to_map(point)
-#			print(point2)
-#			print("\n")
-#			map.erase_cell(1, point2)
-#			map.set_cell(0, point2, 0, Vector2i(0,0), 0)
+# Aka how many damage it deals per second
+var mining_speed = 10
+
+
+func mine(delta: float, player: Player, collision: RayCast2D) -> void:
```

Comment:
I don't know if this coupling of the component with the player is that good. It would be optimal if this component could work for anything like other entites as well i.e future enemies that can mine.  
So one way would be to have an entity as parameter or maybe even better emit a signal.

Reply:
Problem with a signal might be said enemy destroying an ore and the player getting it

To avoid that the signal would in turn need to provide who mined -> parameter needed

Or am I missing something here?